docs/docs/plugin-authoring.md

```diff
@@ -1,15 +1,14 @@
 ---
-title: Plugin Authoring
+title: Plugin authoring
```

Comment:
@m-allanson and @pieh do you remember if there's anything else that needs to be edited about this page? (i.e. common obstacles contributors are running into when they submit plugins?). I remember Mike, you were mentioning recently that you created an issue to help plugin authors signify what version of Gatsby their plugin is compatible with. Or was it the other way around, just helping people vet which plugins work for v2? Either way, wondering if there's something that needs to be edited here in this doc)

Reply:
Oh good idea - here the issue: https://github.com/gatsbyjs/gatsby/issues/7143.

I think we could merge this PR and then make another pass on the plugin docs to mention the Gatsby peerDependency?

Reply:
I'm not sure where it would fit in - but maybe this could mention something about learning from the source code of existing plugins? 

Most plugins are open source and the code is often quite readable - if you're looking to write your own plugin it can be useful to review the source of similar plugins.

Reply:
Thinking further on this - maybe encourage people to write in a style that others can learn from? 

That could be going too far off-track for this doc though. Sometimes people just need to get their stuff done and move on to the next job, so being told how to write your code here could be counter-productive.